vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -2226,10 +2227,43 @@ public void centerAt(Point p, int dx, int dy) {
   /** Ensure that the given region (in map coordinates) is visible */
   public void ensureVisible(Rectangle r) {
     if (scroll != null) {
+      boolean bTriggerRecenter = false;
       final Point p = mapToComponent(r.getLocation());
-      r = new Rectangle(p.x, p.y,
-            (int) (getZoom() * r.width), (int) (getZoom() * r.height));
-      theMap.scrollRectToVisible(r);
+      final Rectangle rCurrent = theMap.getVisibleRect();
+      Rectangle rNorecenter = new Rectangle(0, 0);
+
+      // If r is already visible decide if unit is close enough to
+      // border to justify a recenter
+
+      // Close enough means a strip of the window along the edges whose
+      // width is a % of the edge to center of the window
+
+      // The % is defined in GlobalOptions.CENTER_ON_MOVE_SENSITIVITY
+      final double noRecenterPct = (100 - GlobalOptions.getInstance().centerOnOpponentsMoveSensitivity()) / 100;
```

Comment:
If you want floating point division, you need to ensure one of the operands is a floating point type, say by using `100.0` instead of `100`.